runtime/src/accounts_db.rs

```diff
@@ -17671,4 +17686,402 @@ pub mod tests {
+            }
+        }
+
+        // Verify secondary index list matches the list you tracked above.
+        let largest_keys = accounts_db.accounts_index.get_largest_keys(
+            &AccountIndex::ProgramId,
+            MAX_NUM_LARGEST_INDEX_KEYS_RETURNED,
+        );
+        // Reverse the tracking vector and check for equality
+        test_largest_keys = test_largest_keys.into_iter().rev().collect_vec();
+        assert_eq!(test_largest_keys, largest_keys);
+
+        // Test querries for a partial list and past max return size
+        for i in 0..=MAX_NUM_LARGEST_INDEX_KEYS_RETURNED + 1 {
+            test_largest_keys = accounts_db
+                .accounts_index
+                .get_largest_keys(&AccountIndex::ProgramId, i);
+            if i <= MAX_NUM_LARGEST_INDEX_KEYS_RETURNED {
+                assert_eq!(test_largest_keys.len(), i);
+            } else {
+                assert_eq!(test_largest_keys.len(), MAX_NUM_LARGEST_INDEX_KEYS_RETURNED);
+            }
+        }
+
+        let num_token_accounts_range = Uniform::new_inclusive(1, MAX_TOKEN_ACCOUNTS);
+        let num_mint_accounts_range = Uniform::new_inclusive(NUM_WALLETS, MAX_MINT_ACCOUNTS);
+        let wallet_pubkey_index_range = Uniform::new(0, NUM_WALLETS);
+
+        let num_token_accounts = (&mut num_generator)
+            .sample_iter(num_token_accounts_range)
+            .next();
+        let num_mint_accounts = (&mut num_generator)
+            .sample_iter(num_mint_accounts_range)
+            .next();
+
+        // Make a bunch of SPL Tokens each with some random number of SPL Token Accounts that have the token in them
+        for _i in 0..num_mint_accounts.unwrap_or(NUM_WALLETS) {
+            let mint_pubkey = Pubkey::new_unique();
+            for _j in 0..num_token_accounts.unwrap_or(1) {
+                let owner_pubkey = wallet_pubkeys[(&mut num_generator)
+                    .sample_iter(wallet_pubkey_index_range)
+                    .next()
+                    .unwrap()];
+                let account_pubkey = Pubkey::new_unique();
+                // Add a token account
+                let mut account_data = generic_spl_token_account.data;
+                account_data[generic_spl_token_account.get_mint()]
+                    .copy_from_slice(mint_pubkey.as_ref());
+                account_data[generic_spl_token_account.get_owner()]
+                    .copy_from_slice(owner_pubkey.as_ref());
+                assert_eq!(SPL_TOKEN_ACCOUNT_LENGTH, account_data.len());
+                let token_account = AccountSharedData::from(Account {
+                    lamports: 22222222,
+                    data: Vec::from(account_data),
+                    owner: inline_spl_token::id(),
+                    ..Account::default()
+                });
+                accounts_db.store_for_tests(slot, &[(&account_pubkey, &token_account)]);
+                slot += 1;
+            }
+            // Add the mint
```

Comment:
Now that I see how you're using this... I think you can strip out all the mint stuff.
The mint account doesn't need to be in accountsdb for a token account that refers to it to be added to the secondary index.

Reply:
Added test using the other indexes and the interface mentioned above.